includes/Modules/AdSense.php

```diff
@@ -496,22 +496,52 @@ protected function create_data_request( Data_Request $data ) {
 					return true;
 				};
 			case 'GET:earnings':
-				$dates = $this->date_range_to_dates( $data['dateRange'] ?: 'last-28-days' );
+				$start_date = $data['startDate'];
+				$end_date   = $data['endDate'];
+				if ( ! strtotime( $start_date ) || ! strtotime( $end_date ) ) {
+					$dates = $this->date_range_to_dates( $data['dateRange'] ?: 'last-28-days' );
+					if ( is_wp_error( $dates ) ) {
+						return $dates;
+					}
+
+					list ( $start_date, $end_date ) = $dates;
+				}
 
-				if ( is_wp_error( $dates ) ) {
-					return $dates;
+				$args = array(
+					'start_date' => $start_date,
+					'end_date'   => $end_date,
+				);
+
+				$metrics = $data['metrics'];
+				$metrics = is_array( $metrics ) ? $metrics : explode( ',', $metrics );
+				$metrics = array_filter( array_map( 'trim', $metrics ) );
+				if ( empty( $metrics ) ) {
+					return new WP_Error(
+						'missing_required_param',
+						/* translators: %s: Missing parameter name */
+						sprintf( __( 'Request parameter is empty: %s.', 'google-site-kit' ), 'metrics' ),
+						array( 'status' => 400 )
+					);
+				} else {
+					$args['metrics'] = $metrics;
 				}
```

Comment:
We'll need to make further adjustments to `create_adsense_earning_data_request` to make this work, as this doesn't just pass through `$args`, but creates a new array. Also `metrics` specifically is right now force-set (see related above comment).

Reply:
Updated.